### ISSUE TITLE
Support for union of records

### ISSUE THREAD

**Issue by jchia** [lerouxrgd/rsgen-avro#25]
`rsgen-avro` rejects the following schema with an error `AvroRs error: Unions cannot contain duplicate types`:
```
{
    "type": "record",
    "name": "Foo",
    "fields": [
        {"name": "a", "type": "int"},
        {"name": "b",
         "type":
            [
                {
                    "type": "record",
                    "name": "Bar",
                    "fields": [
                        {"name": "c", "type": "long"}
                    ]
                },
                {
                    "type": "record",
                    "name": "Baz",
                    "fields": [
                        {"name": "d", "type": "int"}
                    ]
                }
            ]
        }
    ]
}
```

Is the schema wrong (I don't think so)? If so, how should it be fixed? Otherwise, `resgen-avro` need so to be fixed. I suspect it is confusedly thinking that `Bar` and `Baz` are the same thing simply because of the `type` of `record`.

The error vanishes when I replace the Baz record block with "int" or "long", but that's not the union I want to express.

**Comment by martin-g** [comment]
Could you please share the panic backtrace ?

**Comment by jchia** [comment]
There was no panic, just a calm error message with exit code of 1. Setting `RUST_BACKTRACE` did not change the output.

**Comment by lerouxrgd** [comment]
The schema looks good indeed. So it's a bug in `avro-rs` crate that cannot parse this schema correctly it seems ...

**Comment by martin-g** [comment]
I've created https://github.com/lerouxrgd/rsgen-avro/pull/26 
It works fine with apache_avro (the successor of avro_rs).

**Comment by martin-g** [comment]
With apache_avro the produced output looks like:

```

#[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize)]
#[serde(default)]
pub struct Baz {
    pub d: i32,
}

impl Default for Baz {
    fn default() -> Baz {
        Baz {
            d: 0,
        }
    }
}

#[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize)]
#[serde(default)]
pub struct Bar {
    pub c: i64,
}

impl Default for Bar {
    fn default() -> Bar {
        Bar {
            c: 0,
        }
    }
}

/// Auto-generated type for unnamed Avro union variants.
#[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize)]
pub enum UnionBarBaz {
    Bar(Bar),
    Baz(Baz),
}

#[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize)]
#[serde(default)]
pub struct Foo {
    pub a: i32,
    pub b: UnionBarBaz,
}

impl Default for Foo {
    fn default() -> Foo {
        Foo {
            a: 0,
            b: UnionBarBaz::Bar(Bar::default()),
        }
    }
}
```

**Comment by lerouxrgd** [comment]
Fixed in `0.11.0`.

Sorry GitHub automatically closed the issue when I merged the related PR, don't hesitate to re-open if something is not correct.
